### ISSUE TITLE
As a `data scientist`, I want to `use pixiedust with IBM Analytics Engine` so that `I can install spark packages` 

### ISSUE THREAD

**Issue by snowch** [pixiedust/pixiedust#492]
As a `data scientist`, I want to `use pixiedust with IBM Analytics Engine` so that `I can install spark packages`

**Comment by DTAIEB** [comment]
@snowch hum... it should work the same with IAE. Are you seeing errors when using installPackage method?

**Comment by snowch** [comment]
@DTAIEB yes. I tried to add the spark avro package.  It installed it without error, but after restarting the kernel, I get:

```
df = spark.read.format("com.databricks.spark.avro").load("twitter.avro")
df.show(5)

Py4JJavaError: An error occurred while calling o108.load.
: org.apache.spark.sql.AnalysisException: Failed to find data source: com.databricks.spark.avro. Please find an Avro package at http://spark.apache.org/third-party-projects.html;
```

I've also tried to install directly using SparkContext.addFile() but couldn't get that working either: https://gist.github.com/snowch/896bec14174afe9e166013691c64bfc7

There is some classpath weirdness when running notebooks via DSX - see this internal ticket: https://github.ibm.com/wdp-chs/tracker/issues/2890

**Comment by DTAIEB** [comment]
The problem is that the IBM Analytics Engines service doesn't add the directory where PixieDust download the packaged jars to the Classpath. Here's the work around:
1. run `pixiedust.installPackage('<package_I_want_to_install>')`
2. use `pixiedust.printAllPackages()` to get the path for the dowloaded jars e.g. /home/xxx/data/libs
3. Go to the Ambari-console (Manage/Launch Console), select Spark2 on the left hand side, then configs, then Advanced spark2-env twistie
4. Add /home/xxx/data/libs/* in Advanced spark-env classpath (see screen shot below)
5. Restart the spark service component

![image](https://user-images.githubusercontent.com/4912140/32904748-1ce05ccc-cac6-11e7-9791-0b2b1ccd1f7e.png)

**Comment by snowch** [comment]
@DTAIEB I have had another user run into this problem today.

Why can't pixiedust be fixed?  It should be possible to detect the environment it is running in and based on the environment take different actions to install libraries?

Alternatively, if that isn't going to happen it should at least give a useful error message to tell users that Pixiedust does not support installing libraries on IAE?

**Comment by DTAIEB** [comment]
@snowch This feature is heavily depending on the environment. To work, we need write access to a directory that is on the JVM path. Spark Service is providing such a directory but not IAE. Without it there is no way to "fix it". 
Sure we could have PixieDust give a useful error message when detecting that the directory is not available, but it is important to understand that this won't fix the issue. A better course of action would be to have IAE provide the directory by default, or if not possible at least provide a simpler option to add this directory to the JVM Path than the ambari steps described above.

**Comment by snowch** [comment]
@DTAIEB can we go with the useful error for now until a solution has been agreed with the IAE team?

**Comment by vabarbosa** [comment]
similar to #730 
should be resolved by #732
